### DISCUSSION TITLE
How to setup /blog route

### DISCUSSION THREAD

**Discussion by joan-medina-roca** [GetPublii/Publii#1466] [Help]
Maybe I'm missing the obvious here.

I try to set up my blog just as in Getpublii.com/blog, where each post appears after /blog/, also authors and tags retain the /blog/ url. Examples: https://getpublii.com/blog/tutorials/ https://getpublii.com/blog/authors/bob-mitro/ .... well just check the blog.

If anyone can guide me in the right direction... Thanks!

**Comment by 3muchfun** [comment]
Under 'Server' on the left hand side, try adding _/blog_  at the website URL, like I did for 'fotos'

<img width="906" alt="Bildschirmfoto 2024-04-02 um 22 15 08" src="https://github.com/GetPublii/Publii/assets/17292904/033e6874-73b4-4ade-aacc-727ae8252801">

**Reply by 3muchfun** [reply]
PS: Before uploading, the folder /blog must exist on the server

**Comment by joan-medina-roca** [comment]
I get your idea, however I already had figured that one out. Doing that, I will be using Publii as a blog system that works just for the /blog section, right?
However, what I try to do is to use Publii across the whole domain.
Create all the pages, all the posts, everything using Publii.
I want to be able to create a route for /blog which displays all the posts, tags and authors keeping the /blog URL. But in the case of pages, which are Hidden posts, not to show the /blog URL. Maybe now my idea is better explained, sorry for the confusion, and thanks for the prompt reply!

**Reply by 3muchfun** [reply]
yes, I see, what you want to achieve. I think, I would set up a website /blog with all the posts etc for your blog ( like website.tld/blog), and a second website which holds the 'pages' and has a link to website.tld

**Reply by joan-medina-roca** [reply]
Yes, this solution would work of course.

However, there must be a way to do it within Publii, just using one website and CMS for everything.

Maybe using helpers or global variables? I'm looking into those, but I have no clue how to use them (I'm not a developer, but I will try).

**Reply by 3muchfun** [reply]
I use several websites/kind of sub-domains, and I do it this way: 

![Bildschirmfoto 2024-04-03 um 11 03 37](https://github.com/GetPublii/Publii/assets/17292904/26cb9788-2958-47f7-8468-9102e018d12f)

**Reply by 3muchfun** [reply]
and I have corresponding folders on the server for uploading the content with Publii

**Reply by 3muchfun** [reply]
but maybe @bobmitro  oder @dziudek have a better solution for your approach

**Reply by joan-medina-roca** [reply]
Yes, I checked out your website and realised that.

However, I want to do it with just one instance of Publii for everything. I think thats how the devs did it at Getpublii.com, and I want to know how.

**Reply by joan-medina-roca** [reply]
(you can check a website I did previously, and all using only Publii. However, for the blog posts I just tagged them as #blog and created a link at /blog showing all the posts with the #blog tag: [lamardeswing.com](https://lamardeswing.com/))

**Comment by candidexmedia** [comment]
> However, I want to do it with just one instance of Publii for everything. I think thats how the devs did it at Getpublii.com, and I want to know how.

I don't think that the entire getpublii site is managed under one Publii instance. I say this because back when we had the old forum, they were using Wordpress for that portion of the site. 

Now, I still believe that the different directories (/blog vs /dev vs /docs) are likely using different instances of Publii because the searches are isolated to each portion of the site. For instance, using the search in the [/docs](https://getpublii.com/docs/) section of the site doesn't provide results from the /blog.

The devs will have to confirm, but I think they are just duplicating their stylesheets and menus between instances (and there have been times when I noticed slight differences between the menus in different sections). That said, maintaining multiple sites this way is tedious.

In any case, I know for sure that directories in URLs currently aren't supported. See related discussions:

- https://github.com/GetPublii/Publii/discussions/1450#discussioncomment-8875920
- https://github.com/GetPublii/Publii/discussions/1413#discussioncomment-8696773
- https://getpublii.com/dev/how-urls-in-publii-works/

**Reply by joan-medina-roca** [reply]
Thanks so much for this in-depth answer! However, yes, it is very tedious to maintain sites in this fashion, that's why I'm looking for a solution to this.

**Reply by candidexmedia** [reply]
@joan-medina-roca You should create/piggyback on some sort of feature request (in Issues perhaps?) because it's [not currently possible to do under one instance out of the box](https://getpublii.com/dev/how-urls-in-publii-works/). I vaguely remember someone in the old forum mentioning doing this for a multilingual site, but it was pretty hacky and also tedious.
